From 8c2223b332cece9f1b4cb84185d2e1d9719af7cd Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Sat, 23 Nov 2024 11:19:31 +0800
Subject: [PATCH 251/280] WIP: dmaengine: pl330: Report DMA_IN_PROGRESS for
 infinitely cyclic

The infinitely cyclic without CPU intervention use the single desc,
so, should always return DMA_IN_PROGRESS to match its status.

Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
Assisted-by: Claude:claude-opus-5-5
Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 drivers/dma/pl330.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/dma/pl330.c b/drivers/dma/pl330.c
index c6bc073d3382..0f41e66fb40b 100644
--- a/drivers/dma/pl330.c
+++ b/drivers/dma/pl330.c
@@ -2583,6 +2583,13 @@ pl330_tx_status(struct dma_chan *chan, dma_cookie_t cookie,
 			default:
 				WARN_ON(1);
 			}
+			/*
+			 * The infinitely cyclic without CPU intervention
+			 * use the single desc, so, should always return
+			 * DMA_IN_PROGRESS to match its status.
+			 */
+			if (desc->cyclic)
+				ret = DMA_IN_PROGRESS;
 			break;
 		}
 		if (desc->last)
-- 
2.43.0

